honu_test.go

```diff
@@ -3,24 +3,32 @@ package honu_test
 import (
 	"fmt"
 	"io/ioutil"
-	"math/rand"
 	"os"
 	"testing"
 	"time"
 
 	"github.com/rotationalio/honu"
 	"github.com/rotationalio/honu/config"
 	pb "github.com/rotationalio/honu/object"
+	"github.com/rotationalio/honu/options"
 	"github.com/stretchr/testify/require"
-	"github.com/syndtr/goleveldb/leveldb"
 )
 
-func setupHonuDB() (*honu.DB, string, error) {
+// Returns a constant list of namespace strings.
+// TODO: Share with engines/leveldb/leveldb_test.go
+func getNamespaces() []string {
+	return []string{
+		"",
+		"basic",
+		"namespace with spaces",
+		"namespace::with::colons",
+	}
+}
```

Comment:
I was trying to essentially keep it as a constant but since go doesn't support constant slices I thought a function would be a decent compromise. I'm not modifying the variable though, so I can change it to a variable.

Reply:
That makes sense - I suppose the primary motivation you're thinking of is that the value should not be changed during the tests and a functional guard is an excellent way to handle that if you need array manipulation without copying. Go constants are a compile-time directive not a runtime directive and that's why arrays, maps, slices, etc. can't be declared as constant since they need to be allocated at runtime. 

My primary concern for the constant here is that it is `static` - e.g. it is declared once for the lifetime of the program. The functional guard means that the slice is allocated for every test. 

One compromise is to make this a static array instead of a slice:

```go
var testNamespaces = [4]string{
		"",
		"basic",
		"namespace with spaces",
		"namespace::with::colons",
	}
```

This prevents users from appending or changing the size of the variable, which helps prevent coding mistakes.